vendor/github.com/hashicorp/nomad/api/jobs.go

```diff
@@ -850,6 +850,9 @@ func (j *Job) Canonicalize() {
 	if j.VaultToken == nil {
 		j.VaultToken = stringToPtr("")
 	}
+	if j.NomadTokenID == nil {
```

Comment:
@tgross just tagging for visibility, its in the API package so I'm guessing it should be in here

Reply:
Thanks. That should have been in https://github.com/hashicorp/nomad/pull/8204 but I guess I `make sync` and then didn't commit it or something.